DragaliaAPI.Database/Entities/DbFortDetail.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using Microsoft.EntityFrameworkCore;
+
+namespace DragaliaAPI.Database.Entities;
+
+[Table("PlayerFortDetail")]
+[Index(nameof(DeviceAccountId))]
+public class DbFortDetail : IDbHasAccountId
+{
+    /// <inheritdoc />
+    public virtual DbPlayer? Owner { get; set; }
+
+    /// <inheritdoc />
+    [Key]
+    [ForeignKey(nameof(Owner))]
+    public required string DeviceAccountId { get; set; }
+
+    [Column("CarpenterNum")]
+    [Required]
+    public int CarpenterNum { get; set; }
+
+    [Column("MaxCarpenterCount")]
+    [Required]
+    public int MaxCarpenterCount { get; set; }
+
+    [Column("WorkingCarpenterNum")]
+    [Required]
+    public int WorkingCarpenterNum { get; set; }
```

Comment:
Just had some thoughts about this DB entity in the meantime

1. Is there any particular reason to track MaxCarpenterCount? It should be the same value (5) for all players, right?
2. Would it be better to calculate WorkingCarpenterNum on-demand by querying buildings with a pending `BuildEndDate`, rather than tracking it as an independent column? Concerned about the possibility of it becoming out-of-sync.

Just some thoughts, not requests for change -- let me know what you think.

Reply:
For this entity, I based myself on the `FortDetail` class that already existed in `Models\Generated\Components.cs`. Didn't think too critically of it at the time, assumed all fields would be required at some point.

1. The maximum should always be 5, correct. I would be in favor of removing the field from the entity and storing it as a constant wherever it makes sense, I believe it would be the actual Controller in this case.
2. The possibility of desync is real, I ran into it multiple times when implementing these endpoints. In theory, if everything works correctly, it _should_ never happen. I'm not opposed to querying all the builds, but I'm concerned about the performance of that. Wouldn't it be too expensive to run such a query every time? Do note that we have to feed this info on the `/fort/get_data` endpoint, which runs every time someone opens the Halidom menu and the Halidom itself, not just when a build is started/finished.

Reply:
Constant in the controller is fine — the config setting is just a wild idea, and I’m not even sure changing it would work; I vaguely remember seeing defined levels in the client data, presumably it has to know how much the upgrade will cost without asking the server. 

For the active count, I’m sure your code works fine, it’s more a concern if changes are made to this later —seems like the sort of thing that could be easily missed. However do you know if this entity appears in update_data_list? If it does, having it as calculated rather than a db entity may make things a bit tricky.

Reply:
Currently, I'm only using the MaxCarpenterNum on the Controller, which is why I suggested that. If a Service makes more sense, I'm fine with moving it, though.

Regarding active count: It does not seem to be included in `update_data_list`. The client gets the info from the `fort_detail`, which is a field part of the response in most, if not all of these endpoints. So I think we would be safe on that end.

I can make those changes so that we can test the performance hit, shouldn't be too much work. If it's bad, we can always go back on them, no big deal.

Reply:
Just pushed these changes and a few more requested in the PR.

About the constant, there is currently no service for the Fort implementation, and I'm unsure if there should be one or not. For now, I've moved the constant to the Repository where it probably makes more sense than the controller.

Reply:
Yeah I think a separate fort service should be good. The usual structure I try to go with is:

- controller parses the request, calls the service, and formats the response
- service performs all the logic behind the response (checking mats, making repository calls)
- repository provides methods to update the database

Reply:
Should be a bit of a refactor work, but I think it makes more sense than the current implementation, I'll get it done